score/checks/checks.go

```diff
@@ -48,7 +49,7 @@ type MetaCheck struct {
 
 type PodCheck struct {
 	ks.Check
-	Fn func(corev1.PodTemplateSpec) scorecard.TestScore
+	Fn func(corev1.PodTemplateSpec, string) scorecard.TestScore
```

Comment:
I think that it would be a nice improvement for type safety and clarity if argument this was a `metav1.TypeMeta` instead of a `string`. This also gives the `APIVersion`, which is already relevant (see my other comment on `probes.go`), and likely will be used again in the future.

```suggestion
	Fn func(corev1.PodTemplateSpec, metav1.TypeMeta) scorecard.TestScore
```